### PR TITLE
Feature/issue 148

### PR DESCRIPTION
**Pull request by rambrus**
*Issue #, if available:*
https://github.com/awslabs/deequ/issues/148

*Description of changes:*
Implemented the `row level completeness` feature as discussed in [issue #148 ](https://github.com/awslabs/deequ/issues/148).
Added the `isCombinedComplete` and `hasCombinedCompleteness` checks to Check class.

To be honest, I'm not convinced that this is the best approach.

I think we have these alternatives to be considered:
- Overload `isComplete` / `hasCompleteness` checks the way you did in `hasUniqueness` Check to be able to support multiple columns
- Do not add a new Check, just provide some predefined Column conditions the way I did in [ColumnCondition](https://github.com/awslabs/deequ/compare/master...rambrus:feature/issue-148#diff-3239736df413d462a02bbcdbef1228f2) object. In this case, the API would look like:
  ```
	.addCheck(Check.satisfies(isEachNonNull(cols), "Multi-Column Completeness"))
  ```
  
I think it's pretty readable, maybe it would be easier to cover these not-so-frequent use cases by providing column conditions instead of Checks. 

However, the question is: do we want to define to these column conditions in the `deequ` project or leave it to the users to do it on client side?
(I think it would be useful to provide some common column conditions in `deequ`.)

@sscdotopen  Please let me know your thoughts on the above.
Thank you,
Robert

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/checks/Check.scala [hunk 1]
```diff
@@ -130,6 +131,36 @@ case class Check(
...
   }
 
+  /**
+    * Creates a constraint that asserts on completion in combined set of columns.
+    *
+    * @param columns Columns to run the assertion on
+    * @param hint A hint to provide additional context why a constraint could have failed
+    * @return
+    */
+  def isCombinedComplete(
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/checks/Check.scala]: rename this to 'areComplete'

File: src/main/scala/com/amazon/deequ/checks/Check.scala [hunk 2]
```diff
@@ -130,6 +131,36 @@ case class Check(
...
+
+  /**
+    * Creates a constraint that assert on completion in combined set of columns.
+    *
+    * @param columns Columns to run the assertion on
+    * @param assertion Function that receives a double input parameter and returns a boolean
+    * @param hint A hint to provide additional context why a constraint could have failed
+    * @return
+    */
+  def hasCombinedCompleteness(
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/checks/Check.scala]: rename this to 'haveCombinedCompleteness'

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package com.amazon.deequ.checks
+
+object ColumnCondition {
+  def isNonNull(col: String): String = {
+    s"$col IS NOT NULL"
+  }
+
+  def isEachNonNull(cols: Seq[String]): String = {
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: Please dont build the predicate via string concatenation, which is error prone. Please use Spark's functions from org.apache.spark.sql.function for constructing this.

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.deequ.checks
+
+object ColumnCondition {
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: Please make this private, as it is only intented for use inside Deequ

File: src/main/scala/com/amazon/deequ/checks/Check.scala (hunk 1 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/Check.scala]: good idea!

File: src/main/scala/com/amazon/deequ/checks/Check.scala (hunk 2 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/Check.scala]: should we include `Combined` in the name? maybe `haveCompleteness` to follow the same convention we used in `areComplete`?

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala (hunk 3 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: all right, will do

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala (hunk 4 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: sure

File: src/main/scala/com/amazon/deequ/checks/Check.scala (hunk 2 above)
Comment (sscdotopen) [reply src/main/scala/com/amazon/deequ/checks/Check.scala]: Yes, good point, lets call it haveCompleteness

Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/Check.scala]: done

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala (hunk 3 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: done

File: src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala (hunk 4 above)
Comment (rambrus) [reply src/main/scala/com/amazon/deequ/checks/ColumnCondition.scala]: done

**Comment by sscdotopen** [comment]
Thank you for the contribution!

**Pull request merged**